---
title: "コンパイラ エラー CS1930"
ms.date: 07/20/2015
ms.prod: .net
ms.technology:
- devlang-csharp
ms.topic: article
f1_keywords:
- CS1930
helpviewer_keywords:
- CS1930
ms.assetid: d28d2334-825c-4ffc-b9e9-f5d61f44d672
caps.latest.revision: 
author: BillWagner
ms.author: wiwagn
ms.openlocfilehash: 1dd166326ff5ba1eec1e3100c6c2d3e9ae39c10c
ms.sourcegitcommit: 83dd5ec003e788ccb3e33f3412a7af39ae347646
ms.translationtype: MT
ms.contentlocale: ja-JP
ms.lasthandoff: 03/15/2018
---
# <a name="compiler-error-cs1930"></a>コンパイラ エラー CS1930
範囲変数 'name' は既に宣言されています。  
  
 クエリ式が終了するまで、クエリ式の範囲変数がスコープ内に存在します。 そのため、一意の識別子が必要です。  
  
## <a name="to-correct-this-error"></a>このエラーを解決するには  
  
1.  クエリ式で導入された各範囲変数に、一意の名前を指定します。  
  
## <a name="example"></a>例  
 次の例では、識別子 `num` が、 `from` 句の範囲変数および `let` 句によって導入された範囲変数に使用されるため、CS1930 が生成されます。  
  
```csharp  
// cs1930.cs  
using System.Linq;  
class Program  
{  
    static void Main()  
    {  
        int[] nums = { 0, 1, 2, 3, 4, 5 };  
        var query = from num in nums  
                    let num = 3 // CS1930  
                    select num;   
    }  
}  
```  
  
## <a name="see-also"></a>参照  
 [LINQ クエリ式](../../csharp/programming-guide/linq-query-expressions/index.md)
